hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/io/ECKeyOutputStream.java

```diff
@@ -660,6 +515,45 @@ private ByteBuffer[] getParityBuffers() {
       return parityBuffers;
     }
 
+    private long getDataSize() {
+      return Arrays.stream(dataBuffers)
+          .mapToInt(Buffer::position)
+          .asLongStream()
+          .sum();
+    }
+
+    private boolean isPartialStripe() {
+      return dataBuffers[0].position() > 0 &&
+          dataBuffers[dataBuffers.length - 1].position() < cellSize;
+    }
+
```

Comment:
Why ECChhunkBuffers do encode functionality. This class should be simple, only stores buffers and give buffers when needed. Do the business logic outside of this class.